lib/Logger/LogMacros.h

```diff
@@ -58,16 +58,35 @@
 #include "Logger/Logger.h"
 #include "Logger/LoggerStream.h"
 
-////////////////////////////////////////////////////////////////////////////////
-/// @brief logs a message for a topic
-////////////////////////////////////////////////////////////////////////////////
 
 #define ARANGO_INTERNAL_LOG_HELPER(id)                        \
   ::arangodb::Logger::LINE(__LINE__)                          \
   << ::arangodb::Logger::FILE(__FILE__)                       \
   << ::arangodb::Logger::FUNCTION(__FUNCTION__)               \
   << ::arangodb::Logger::LOGID((id))
 
+#define ARANGODB_UNCONDITIONALLY_BUILD_LOG_MESSAGES 1
+#if ARANGODB_UNCONDITIONALLY_BUILD_LOG_MESSAGES
+// these versions *always* generate the log string, only to discard it
+// just before logging them. This is good for ASAN or coverage.
+// DONT use in production.
+#define LOG_TOPIC(id, level, topic)                                     \
+  ::arangodb::LoggerStream() << (::arangodb::LogLevel::level)           \
+                             << (topic)                                 \
+                             << ARANGO_INTERNAL_LOG_HELPER(id)
+
+#define LOG_TOPIC_IF(id, level, topic, cond)                            \
+  ::arangodb::LoggerStream()                                            \
+  << ::arangodb::LogLevel::TRACE                                        \
+  << (topic)                                                            \
+  << ARANGO_INTERNAL_LOG_HELPER(id)                                     \
+  << (true && cond)
```

Comment:
Macro parameters containing arbitrary expressions should be put inside parentheses.
Additionally, I don't see the value of `true && cond` here. Shouldn't it be just `(cond)`?
```suggestion
  << (cond)
```